repo/tuf_on_ci/signing_event.py

```diff
@@ -225,6 +225,67 @@ def update_targets(verbose: int, push: bool) -> None:
     sys.exit(0 if updated_targets else 1)
 
 
+@click.command()
+@click.option("-v", "--verbose", count=True, default=0)
+@click.option("--push/--no-push", default=True)
+@click.argument("roles", nargs=-1, metavar="role")
+def online_sign_targets(verbose: int, push: bool, roles: list[str]) -> None:
+    """Sign targets metadata with KMS during a signing event.
+
+    Attempts to sign specified targets roles metadata with online signing keys.
+
+    Return value is 0 if any targets metadata was signed.
+    """
+    logging.basicConfig(level=logging.WARNING - verbose * 10)
+
+    if not os.path.exists("metadata/root.json"):
+        sys.exit(1)
+
+    event_name = _git(["branch", "--show-current"]).stdout.strip()
+    head = _git(["rev-parse", "HEAD"]).stdout.strip()
+
+    signed_targets = []
+    repo = CIRepository("metadata")
+    for role in roles:
+        if role in ["root", "timestamp", "snapshot"]:
+            raise ValueError(f"Non-targets role '{role}' cannot be signed")
+
+        if repo.is_signed(role):
+            click.echo(f"Signing not required for role '{role}'")
+            continue
```

Comment:
this kind of also encodes the assumption that there is a single signer: is_signed() returns true if role is signed by threshold of sigs (and is not in signing period) -- I suppose that's fine as long as online signing always means 1 signer